lib/p2p/Peer.ts

```diff
@@ -169,6 +218,98 @@ class Peer extends EventEmitter {
     });
   }
 
+  private initStall = (): void => {
+    assert(!this.destroyed);
+    assert(!this.stallTimer);
+
+    this.stallTimer = setInterval(this.maybeTimeout, Peer.STALL_INTERVAL);
+  }
+
+  private initHello = async () => {
+    const packet = this.sendHello();
+
+    if (!this.handshakeState) {
+      await this.wait(packet.responseType);
+      assert(this.handshakeState);
+    }
+  }
+
+  private finalizeOpen = (): void => {
+    assert(!this.destroyed);
+
+    // Setup the ping interval
+    this.pingTimer = setInterval(this.sendPing, Peer.PING_INTERVAL);
+  }
+
+  /**
+   * Wait for a packet to be received from peer. Executed on timeout or once packet is received
+   */
+
+  private wait = (packetType: PacketType) => {
+    return new Promise((resolve, reject) => {
+      const entry = this.getOrAddResponseEntry(packetType);
+      entry.addJob(resolve, reject);
+    });
+  }
+
+  /**
+   * Potentially timeout peer if it hasn't responded
+   */
+
+  private maybeTimeout = () => {
```

Comment:
I think a method name like `checkTimeout` or `checkForTimeout` would be more descriptive.

Reply:
Fixed.